src/app/pages/projects/projects.component.ts

```diff
@@ -1,15 +1,53 @@
-import { Component, OnInit } from '@angular/core';
+import {AfterViewInit, Component, OnInit, ViewChild} from '@angular/core';
+import {MatTableDataSource} from '@angular/material/table';
+import {ModalAddProjectComponent} from "./modal-add-project/modal-add-project.component";
+import {MatSnackBar} from "@angular/material/snack-bar";
+import {Project, ProjectInterface} from "../../models/project";
+import {ProjectHttpService} from "../../services/project-http.service";
+import {AuthService} from "../../services/auth.service";
 
 @Component({
-  selector: 'ngx-projects',
+  selector: 'ngx-app-projects',
   templateUrl: './projects.component.html',
-  styleUrls: ['./projects.component.scss']
+  styleUrls: ['./projects.component.scss'],
 })
-export class ProjectsComponent implements OnInit {
 
-  constructor() { }
+export class ProjectsComponent implements AfterViewInit {
+  userId : number;
 
-  ngOnInit() {
+  displayedColumns: string[] = ["id", "title", "edit"];
+  public dataSource = new MatTableDataSource<ProjectInterface>([]); // output table
+
+  @ViewChild(ModalAddProjectComponent, {static: false})
+  private modalAddProject: ModalAddProjectComponent;
+
+  constructor(
+    private projectHttpService: ProjectHttpService,
+    private authService: AuthService,
+  ) { }
+
+  ngAfterViewInit() {
+    this.authService.user$.subscribe(r => {
+      this.userId = r.id;
+    });
+    this.getProjects();
   }
 
+  getProjects() {
+    const self = this;
```

Comment:
this line is not needed. `this` keyword should work, otherwise something wrong with setup